File: lib/scala/profiling-utils/src/main/scala/org/enso/profiling/NoopSampler.scala
package org.enso.profiling

import java.util.concurrent.Executor

import scala.concurrent.duration.FiniteDuration

/** Sampler that does nothing. */
final class NoopSampler extends MethodsSampler {

  /** @inheritdoc */
  override def start(): Unit = ()

  /** @inheritdoc */
  override def stop(): Unit = ()

  /** @inheritdoc */
  override def stop(delay: FiniteDuration)(implicit ec: Executor): Unit = ()
}
object NoopSampler {

  /** Create an instance of [[NoopSampler]]. */
  def apply(): NoopSampler =
    new NoopSampler
}

File: lib/scala/profiling-utils/src/main/java/org/enso/profiling/sampler/NoopSampler.java
package org.enso.profiling.sampler;

import java.time.Instant;

/** Sampler that does nothing. */
final class NoopSampler implements MethodsSampler {

  @Override
  public void start() {}

  @Override
  public void close() {}

  @Override
  public void log(Instant at, String message) {}
}
